Refreshed the setting up section in the testing dev doc (#7649)
jenkins-infra/jenkins.io: content/doc/developer/testing/index.adoc

5. link:https://htmlunit.sourceforge.io/[HtmlUnit] support, making it simple to test interaction with the web UI and other HTTP calls.

== Setting Up

> This section covers the things to get started with testing.

Reviewer: ```suggestion
This section covers what to test when getting started.
```
Suggestion to adjust the wording a bit, removes the ambiguous "the things".
